lib/pry-doc/pry_ext/show_source_with_c_internals/c_file.rb

```diff
@@ -0,0 +1,49 @@
+# Data looks like:
+#
+# thread.c,11784
+# static VALUE rb_cThreadShield;<\x7f>86,2497
+# static VALUE sym_immediate;<\x7f>88,2529
+# static VALUE sym_on_blocking;<\x7f>89,2557
+
+# First line is the name of the file
+# Following lines are the symbols followed by line number with char 127 as separator.
+module Pry::CInternals
+  class CFile
+    SourceLocation = Struct.new(:file, :line, :original_symbol)
+
+    # Used to separate symbol from line number
+    SYMBOL_SEPARATOR = "\x7f"
+
+    attr_accessor :symbols, :file_name
+
+    def self.from_str(str)
+      new(str).tap(&:process_symbols)
```

Comment:
We never call `CFile.new`, so we can technically invoke `process_symbols` inside `CFile#initialize`. Thus, `CFile.new(str)` will simplify the API (no need to learn a new method). As of now `process_symbols` is a public method, but it can easily be a private one.

Reply:
@kyrylo i don't like to do anything inside `#initialize` methods aside from initialize things, not actually do any real work